src/commands/install-packages.yml

```diff
@@ -11,11 +11,17 @@ parameters:
     type: string
     default: "~/project"
     description: Path to the directory containing your package.json file. Not needed if package.json lives in the root.
-  override-ci:
-    type: boolean
-    default: false
+  override-ci-command:
+    description: |
+      By default, packages will be installed with "npm ci" or "yarn install --frozen-lockfile".
+      Supply a custom package installation command, with any additional flags needed.
+    type: string
+    default: ""
+  cache-path:
     description: |
-      By default, modules will be installed with "npm ci" or "yarn install --frozen-lockfile", this override will use the standard "install" command.
+      By default, this orb will utilize 'npm ci' and cache the '$HOME/.npm' directory. Override which path to cache with this parameter.
+    type: string
+    default: $HOME/.npm
```

Comment:
Will`$HOME` be expanded when used in the `save_cache` path? If not, perhaps `~/.npm` would work?

Reply:
This something that was talked about on the Go Orb as well, I think it would be better to hardcode this to be `/home/circleci` to align with that Orb. I _believe_ that the $HOME expansion depends on the Docker image being used, and does work in most cases.

Reply:
Variables like `$HOME` will:

1. As Gabe said, be expanded depending on the Docker image if it's in a normal run step.
2. Won't be expanded on special steps such as `save_cache`.

I do believe that `~` does work on special steps though it won't hurt to test it to make sure.

---

Lastly, I feel like this is not documented. This should be, in Docs right? If we agree, I'll go ahead and make a PR.

Reply:
The expansion appears to work.

Reply:
Resolving but please let me know if there is an issue.

Reply:
I did some more testing on this and found that the caching doesn't work properly when using $HOME/.npm . The good news is that it works with ~/.npm !

Here is a build link:
https://app.circleci.com/pipelines/github/lokst/circleci-sandbox/2579/workflows/8824cf38-116e-4079-a805-b079bb25225f

When $HOME/.npm is used, in https://app.circleci.com/pipelines/github/lokst/circleci-sandbox/2579/workflows/8824cf38-116e-4079-a805-b079bb25225f/jobs/6463, we can see that the restored cache size is very small:

```
Found a cache from build 6461 at node-deps-v1-node_orb_may20-taPszJYvOWuYzIG2gWYpvJ1lt1ltOlXcowPX25zknFU=
Size: 32 B
```

I also used `override-ci-command` to add the `--verbose` flag to `npm ci`, and verified that when $HOME/.npm is used, packages are being downloaded:

```
Running override package installation command:
npm info it worked if it ends with ok
npm verb cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'ci', '--verbose' ]
npm info using npm@6.14.4
npm info using node@v12.17.0
npm verb npm-session 0d7141fbeacf1514
npm info prepare initializing installer
npm verb prepare starting workers
npm verb prepare installation prefix: /home/circleci/project
npm verb prepare using package-lock.json
npm verb checkLock verifying package-lock data
npm info prepare Done in 0.083s
npm verb extractTree extracting dependencies to node_modules/
npm http fetch GET 200 https://registry.npmjs.org/gulp/-/gulp-4.0.0.tgz 520ms
npm http fetch GET 200 https://registry.npmjs.org/nextjs/-/nextjs-0.0.3.tgz 510ms
npm http fetch GET 200 https://registry.npmjs.org/gulp-cli/-/gulp-cli-2.0.1.tgz 523ms
npm http fetch GET 200 https://registry.npmjs.org/react/-/react-16.13.1.tgz 512ms
npm http fetch GET 200 https://registry.npmjs.org/gulp-mocha/-/gulp-mocha-6.0.0.tgz 524ms
npm http fetch GET 200 https://registry.npmjs.org/http/-/http-0.0.0.tgz 530ms
npm http fetch GET 200 https://registry.npmjs.org/wtfnode/-/wtfnode-0.7.0.tgz 525ms
npm http fetch GET 200 https://registry.npmjs.org/why-is-node-running/-/why-is-node-running-1.2.3.tgz 527ms
npm http fetch GET 200 https://registry.npmjs.org/jest/-/jest-26.0.1.tgz 527ms
```

When using ~/.npm, as shown in https://app.circleci.com/pipelines/github/lokst/circleci-sandbox/2579/workflows/8824cf38-116e-4079-a805-b079bb25225f/jobs/6464, the restored cache size is a lot bigger, as expected (I had specified many sizeable packages to the package.json, like `react-native`, `next` and `jest` to make the difference from caching more visible)

```
Found a cache from build 6462 at node-deps-custom-cache-path-node_orb_may20-taPszJYvOWuYzIG2gWYpvJ1lt1ltOlXcowPX25zknFU=
Size: 149 MiB
```

And we can also see that there is a lack of `npm http fetch` logs in the step that runs `npm ci --verbose`.

Reply:
Thank you stella! Resolved: https://app.circleci.com/pipelines/github/CircleCI-Public/node-orb/541/workflows/144a7585-c51c-43f3-8743-efbcb50ab97d